sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/clangtidy/CxxClangTidySensor.java

```diff
@@ -0,0 +1,119 @@
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.cxx.clangtidy;
+
+import java.io.File;
+import java.util.Scanner;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugins.cxx.CxxLanguage;
+import org.sonar.plugins.cxx.utils.CxxMetrics;
+import org.sonar.plugins.cxx.utils.CxxReportSensor;
+
+/**
+ * Sensor for clang-tidy
+ */
+public class CxxClangTidySensor extends CxxReportSensor {
+  public static final Logger LOG = Loggers.get(CxxClangTidySensor.class);
+  public static final String REPORT_PATH_KEY = "sonar.cxx.clangtidy.reportPath";
+
+  /**
+   * {@inheritDoc}
+   */
+  public CxxClangTidySensor(Settings settings) {
+    super(settings, CxxMetrics.CLANGTIDY);
+  }
+
+  @Override
+  protected String reportPathKey() {
+    return REPORT_PATH_KEY;
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor.onlyOnLanguage(CxxLanguage.KEY).name("CxxClangTidySensor");
+  }
+  
+  @Override
+  protected void processReport(final SensorContext context, File report) {
+    LOG.debug("Parsing clang-tidy report");
+
+    try (Scanner scanner = new Scanner(report, "UTF-8")) {
+      // E:\Development\SonarQube\cxx\sonar-cxx\sonar-cxx-plugin\src\test\resources\org\sonar\plugins\cxx\reports-project\clang-tidy-reports\..\..\cpd.cc:76:20: warning: ISO C++11 does not allow conversion from string literal to 'char *' [clang-diagnostic-writable-strings]
+      // case 1: return "one";
+      // <path>:<line>:<column>: <level>: <message> [<checkname>]
+      // relative paths
+      Pattern p = Pattern.compile("([^:]+):([0-9]+):([0-9]+): ([^:]+): ([^]]+) \\[([^]]+)\\]");
+      Pattern p_drive_letter = Pattern.compile("([^:]+):([^:]+):([0-9]+):([0-9]+): ([^:]+): ([^]]+) \\[([^]]+)\\]");
+
+      while (scanner.hasNextLine()) {
+        String line = scanner.nextLine().trim();
+        if (line.startsWith("/") || line.charAt(1) != ':') {
```

Comment:
actaully its not about windows, or linux.  the problem is windows absolute paths had one more ":" detecting linux or windows wound not work since in windows with relative paths the number of ":" are the same as in linux (absolute or relative).

So the special case is absolute path in windows only, that needs to use the different reg x.